### PR TITLE
Support cancelling previous ci actions

### PR DESCRIPTION
**Pull request by zuston**
### What changes were proposed in this pull request?
Support cancelling previous ci actions when new commits are submitted.

### Why are the changes needed?
I hope the CI actions can always test the latest commits code and cancel the previous ci when new commits are submitted.

Refer to: https://stackoverflow.com/questions/66335225/how-to-cancel-previous-runs-in-the-pr-when-you-push-new-commitsupdate-the-curre

### CODE REVIEW

**Comment by kaijchen** [comment]
> @kaijchen Could u help check this feature? I dont know how to check its effect

Can you try push new empty commits in this PR?

**Comment by kaijchen** [comment]
For reference, this works in Apache Ozone

```yaml
concurrency:
  group: ci-${{ github.event.pull_request.number || github.sha }}
  cancel-in-progress: ${{ github.event_name == 'pull_request' }}
```

**Comment by zuston** [comment]
> > @kaijchen Could u help check this feature? I dont know how to check its effect
> 
> Can you try push new empty commits in this PR?

It works.

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -21,6 +21,11 @@ name: build
 
 on: [push, pull_request]
 
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
+  cancel-in-progress: true
```
Comment (kaijchen) [inline .github/workflows/build.yml:26]: Please change this line to only cancel previous PR actions.

```suggestion
  cancel-in-progress: ${{ github.event_name == 'pull_request' }}
```

**Comment by kaijchen** [comment]
> It works.

Great.

I think it's better to wait all CI on master to finish.
So we know where it starts to fail when something goes wrong.

File: .github/workflows/build.yml (hunk 1 above)
Comment (kaijchen) [reply .github/workflows/build.yml]: Probably this is better.

```suggestion
  cancel-in-progress: ${{ github.ref != 'refs/heads/master' }}
```

**Comment by zuston** [comment]
> > It works.
> 
> Great.
> 
> I think it's better to wait all CI on master to finish. So we know where it starts to fail when something goes wrong.

Make sense.

File: .github/workflows/build.yml (hunk 1 above)
Comment (kaijchen) [reply .github/workflows/build.yml]: > Probably this is better.

I realized we need CI history of release branches or dev branches for big tasks.
So let's keep it as `${{ github.event_name == 'pull_request' }}`

**Review by kaijchen — APPROVED** [review]
LGTM, thanks @zuston.

There is one extra blank line, nevermind.

**Review by jerqi — APPROVED** [review]
LGTM, thanks @zuston @kaijchen

**Pull request merged**